pool-creators/selecting-an-oracle.md

```diff
@@ -2,11 +2,54 @@
 
 Oracles are smart contracts that publish off-chain data on-chain. Blend pools use price oracles to get the prices of their assets.\
 \
-Pool creators must set a pool's oracle contract when they create a pool. This must be a single contract that can report prices for all assets in the pool.
+Pool creators must set a pool's oracle contract when they create a pool. This must be a single contract that can report prices for all assets in the pool.\
 
-Oracles must conform to the SEP-40 oracle standard\
+**Oracles CANNOT be changed after a pool is created, so please be very careful when selecting an oracle for a pool.**
+
+Oracles must support the "lastprice" and "decimals" functions on the SEP-40 oracle standard\
 [https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0040.md](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0040.md)
 
+### Test the Oracle
+
+It is required to verify that "lastprice" works for all potential reserves the pool will maintain. Please note that the Blend pool will always invoke the oracle with `Stellar::Asset({contract_id})` to fetch last price.
+
+The oracle can be tested by using the `PoolOracle` object provided via the [Blend JS SDK](https://github.com/blend-capital/blend-sdk-js).
+
+```js
+import { PoolOracle } from '@blend-capital/blend-sdk';
+
+// allow logging of bigints and maps
+const replacer = (key, value) => {
+  if (value instanceof Map) {
+    return {
+      dataType: "Map",
+      value: Array.from(value.entries()),
+    };
+  } else if (typeof value == "bigint") {
+    return {
+      dataType: "BigInt",
+      value: value.toString(),
+    };
+  } else {
+    return value;
+  }
+};
+
+const network = {
+  passphrase: 'Public Global Stellar Network ; September 2015',
+  rpc: 'some-rpc-url'
+};
+
+// the contract address of the oracle
+const oracle_id = "C...";
+// the contract addresses of the assets being included as reserves
+const assets = ["C...", "C...", "C..."];
+
+const oracle = await PoolOracle.load(network, oracle_id, assets);
+// validate results
+console.log(JSON.stringify(oracle, replacer, 2));
+```
+
 ### Oracle Adapters
 
 When a generalized oracle is insufficient for a pool, i.e. it doesn't support all assets the pool needs prices for, the pool creator may need to use an oracle adaptor. An oracle adaptor is a custom oracle contract that uses custom logic to aggregate multiple oracle feeds or impose desired behavior, such as reporting TWAP prices.
```

Comment:
", i.e. it doesn't support all assets the pool needs prices for," -> (i.e. it doesn't support all assets the pool needs prices for)